src/differentials.jl

```diff
@@ -239,7 +293,19 @@ struct Thunk{F} <: AbstractThunk
     f::F
 end
 
+"""
+    @thunk body
+
+Returns `Thunk(() -> body)`, except for when `body` is a call to [`Wirtinger`](@ref) or [`ComplexGradient`](@ref).
+In this case, it is equivalent to `Wirtinger(@thunk(primal), @thunk(conjugate))` / `ComplexGradient(@thunk primal)`.
```

Comment:
Thought about whether this might be too magical, but this makes implementing the correct behavior for `@scalar_rule` much easier.

Reply:
It is pretty magical.

Also it does des't work if you use 
```
@thunk(ChainRulesCore.Wirtinger(..., ..)

# or 
@thunk(ChainRules.Wirtinger(...,..))

# or
const CR = ChainRulesCore
@thunk(CR.Wirtinger(..,...))

# or 
const Wirt = Wirtinger
@thunk(Wirt(...,...))
```

Reply:
Yes, it also doesn't work, if you do any arithmetic like `2 * Wirtinger(a, b)`. Might be good enough for `@scalar_rule`, though?

Reply:
We could make `(::Thunk)()` error, if `AbstractWirtinger` is returned and otherwise require users to manually implement their rule.

Reply:
Even thought about using IRTools.jl, but I think, we're glad, we just got rid of Cassette ^^

Reply:
If it is just for `@scalar_rule`
then I think it is fine, `@scalar_rule` is already a little DSL with this limitation.

But I them maybe let's not do it in `@thunk` and let's have an internal function for it.

I have been thinking we probably want to Change  the way `@scalar_rule` uses `@thunk` anyway as I think it creatate `@thunk(One())` which is rather inefficient vs `One()`